### DISCUSSION TITLE
Visualize debug output

### DISCUSSION THREAD

**Discussion by sregg** [callstack/react-native-testing-library#755] [Q&A]
Has anyone tried to "visualize" the JSON representation of a component/screen (what `.debug()` returns) ?
I found https://github.com/jest-community/jest-snapshots-svg but it seems abandoned.

**Comment by prometheas** [comment]
I could see this being useful for some more complex views, though my "knee-jerk" rhetorical question in response here (without having even a minute sense of your use case) is whether the perceived "need" for this mightn't turn out to be a "smell" that you might benefit from the decomposition of potentially-"over"-complicated components.

That said, sometimes your components are consuming third-party components, so you may not even have the option of considering such decomposition (assuming, of course, your desire for this viz is indeed driven by complexity / size of output).

**Reply by sregg** [reply]
We're using testing-lib as a replacement for e2e testing like Detox.
They're much faster and we are able to mock the API using MSW.
You can test a single component with testing-lib but you can also render the entire <App/> and navigate through it just like a user would, or am e2e would.
In that case, especially because react-nav comps are not unmounted, the view tree grows pretty quickly.
I wonder if we're the only ones using testing-lib at that level...
